activity_calendar/feeds.py

```diff
@@ -0,0 +1,237 @@
+        return item.description
+
+    def item_start_datetime(self, item):
+        # Convert to Europe/Amsterdam to ensure daylight saving time is accounted for in recurring events
+        return item.start_date.astimezone(timezone.get_current_timezone())
+    
+    def item_end_datetime(self, item):
+        # Convert to Europe/Amsterdam to ensure daylight saving time is accounted for in recurring events
+        return item.end_date.astimezone(timezone.get_current_timezone())
+    
+    def item_created(self, item):
+        return item.created_date
+    
+    def item_updateddate(self, item):
+        return item.last_updated_date
+
+    def item_timestamp(self, item):
+        # When the item was generated, which is at this moment!
+        return timezone.now()
+
+    def item_link(self, item):
+        # There is no special page for the activity
+        return ""
+
+    def item_location(self, item):
+        return item.location
+    
+    def item_transparency(self, item):
+        # Items marked as "TRANSPARENT" show up as 'free' in busy time searches
+        # Items marked as "OPAQUE" show up as 'busy' in busy time searches.
+        return "TRANSPARENT"
+
+    # Recurrence rules for dates to include
+    # E.g. repeat the activity every 3rd wednesday of the month, repeat
+    # every 2 weeks, etc.
+    def item_rrule(self, item):
+        if item.recurrences:
+            rules = []
+            for rule in item.recurrences.rrules:
+                rules.append(build_rrule_from_recurrences_rrule(rule))
+            return rules
+
+    # Recurrence rules for dates to exclude
+    def item_exrule(self, item):
+        if item.recurrences:
+            rules = []
+            for rule in item.recurrences.exrules:
+                rules.append(build_rrule_from_recurrences_rrule(rule))
+            return rules
+
+    # Dates to include for recurrence rules
+    def item_rdate(self, item):
+        if item.recurrences:
+            return item.recurrences.rdates
+
+    # Dates to exclude for recurrence rules
+    def item_exdate(self, item):
+        if item.recurrences:
+            # Each EXDATE's time needs to match the event start-time, but they default to midnigth in the widget!
+            # Since there's no possibility to select the time in the UI either, we're overriding it here
+            # and enforce each EXDATE's time to be equal to the event's start time
+            event_start_time = item.start_date.astimezone(timezone.get_current_timezone()).time()
+            item.recurrences.exdates = list(map(lambda dt:
+                    timezone.get_current_timezone().localize(
+                        datetime.combine(timezone.localtime(dt).date(),
+                        event_start_time)
+                    ),
+                    item.recurrences.exdates
+            ))
+            return item.recurrences.exdates
+
+    # RECURRENCE-ID
+    def item_recurrenceid(self, item):
+        return None
```

Comment:
Looking at the monkey patch at line 15, what does this do and is it needed?

Reply:
At the moment is is not. However, once we support occurrence-editing of a recurring event, then we need to be able to define a RECURRENCE-ID. I added this monkey-patch before I decided that I was splitting event-editing and event-viewing/ical into two separate PRs as to not make them insanely large.
I never bothered to remove it as it was going to be in the next PR anyways.